src/core/manager.c

```diff
@@ -3163,17 +3183,21 @@ void manager_check_finished(Manager *m) {
         if (m->exit_code != MANAGER_OK)
                 return;
 
-        /* For user managers, send out READY=1 as soon as we reach basic.target */
-        if (MANAGER_IS_USER(m) && !m->ready_sent) {
-                Unit *u;
+        if (!m->ready_sent) {
+                /* For user managers, send out READY=1 as soon as we reach basic.target */
+                if (MANAGER_IS_USER(m)) {
+                        Unit *u;
 
-                u = manager_get_unit(m, SPECIAL_BASIC_TARGET);
-                if (u && !u->job) {
-                        sd_notifyf(false,
-                                   "READY=1\n"
-                                   "STATUS=Reached " SPECIAL_BASIC_TARGET ".");
-                        m->ready_sent = true;
-                }
+                        u = manager_get_unit(m, SPECIAL_BASIC_TARGET);
+                        if (u && !u->job) {
+                                sd_notifyf(false,
+                                           "READY=1\n"
+                                           "STATUS=Reached " SPECIAL_BASIC_TARGET ".");
+                        }
+                } else
+                        log_taint_string(m);
+
+                m->ready_sent = true;
```

Comment:
Should not set this here for system manager, that is `! MANAGER_IS_USER(m)`. If set this here, then `READY=1` notification will not be sent.